gunicorn/workers/ggevent.py

```diff
@@ -117,6 +121,19 @@ def run(self):
         except Exception:
             pass
 
+    def _wait_for_keepalive(self, servers):
+        # Retrieve all active greenlets and repeatedly check, until the keepalive period ends,
+        # if any of those greenlets are still active. If none are active, exit the loop.
+
+        greenlets = {id(greenlet) for server in servers for greenlet in server.pool.greenlets}
```

Comment:
When this line is executed, self.alive is already False, which means that all keepalive connections are either closed or are being handled by the existing greenlets. No new keepalive connections will be created until the worker restarts.
Therefore, if the greenlets found in this line have finished, we do not need to wait any longer, as we are assured that there are no more active keepalive connections.